src/wasi/Wasi.cpp

```diff
@@ -108,12 +109,92 @@ WASI::WASI()
     init_options.argc = 0;
     init_options.argv = nullptr;
     init_options.envp = nullptr;
-    init_options.preopenc = 0;
+    if (shareHostEnv) {
+        std::vector<std::string> variables = {
+            // Common Unix environment variables.
+            "PATH",
+            "LIB",
+            "PWD",
+            "SHELL",
+            "TERM",
+            "HOME",
+            "LOGNAME",
+            "HOSTNAME",
+            "UID",
+            "TEMP",
+            "EDITOR"
+            "LANG",
+            "LC_TIME",
+            // Common Windows environment variables.
+            "APPDATA",
+            "USERPROFILE",
+            "USERDOMAIN",
+            "windir",
+            "ProgramFiles",
+            "OS"
+            "LOCALAPPDATA",
+            "SystemRoot",
+            "SystemDrive",
+        };
+
+        char** envp = (char**)malloc(sizeof(char**) * variables.size());
+
+        size_t j = 0;
+        for (size_t i = 0; i < variables.size(); i++) {
+            if (getenv(variables[i].c_str()) == nullptr) {
+                continue;
+            }
+
+            size_t length = strlen(variables[i].c_str());
+            length += strlen(getenv(variables[i].c_str()));
+            length += 2;
+            envp[j] = (char*)malloc(length);
+            strcpy(envp[j], variables[i].c_str());
+            envp[j][variables[i].size()] = '=';
+            strcpy(envp[j] + variables[i].size() + 1, getenv(variables[i].c_str()));
+            envp[j][length - 1] = '\0';
+            j++;
+        }
+
+        while (j < variables.size()) {
+            envp[j] = nullptr;
+            j++;
+        }
+
+        init_options.envp = const_cast<const char**>(envp);
+    }
+    init_options.preopenc = preopenDirs.size();
+    if (init_options.preopenc > 0) {
+        init_options.preopens = reinterpret_cast<uvwasi_preopen_t*>(calloc(init_options.preopenc, sizeof(uvwasi_preopen_s)));
+    }
+    for (unsigned i = 0; i < preopenDirs.size(); i++) {
+        init_options.preopens[i].mapped_path = preopenDirs[i].mapped_path;
+        init_options.preopens[i].real_path = preopenDirs[i].real_path;
+    }
```

Comment:
For this case, `preopenDirs` already contains all the necessary information about directory mapping.
So, it seems that simply assigning the `preopenDirs` into `init_options.preopens` is possible like below
```c++
init_options.preopens = preopenDirs.data();
```
What do you think about this?

Reply:
You are right and it is also easier to understand so I changed it to be this way.